include/engine/routing_algorithms/shortest_path.hpp

```diff
@@ -2,22 +2,464 @@
 #define SHORTEST_PATH_HPP
 
 #include "engine/algorithm.hpp"
+#include "engine/datafacade.hpp"
 #include "engine/routing_algorithms/routing_base.hpp"
-#include "engine/search_engine_data.hpp"
+
 #include "util/typedefs.hpp"
 
+#include <boost/assert.hpp>
+#include <boost/optional.hpp>
+#include <memory>
+
 namespace osrm
 {
 namespace engine
 {
 namespace routing_algorithms
 {
 
+namespace
```

Comment:
Hm. With this way, we need to put all external/offline source code into osrm repo, but it's not allowed for us. Or make any other code hacks ...